test/functional/terminal/buffer_spec.lua

```diff
@@ -255,8 +256,11 @@ describe(':terminal buffer', function()
     command('bdelete!')
   end)
 
-  it('handles wqall', function()
+  it("won't close while a job is running (without !)", function()
     eq('Vim(wqall):E948: Job still running', exc_exec('wqall'))
```

Comment:
would also add asserts for :bdelete and :quit here. Their implementations are all slightly different.

Reply:
`:quit` already works without any warning or error, which is probably a bug.

Reply:
fixing the inverse is a different topic i guess, but we should still add the :quit case here.

Reply:
How should it be added though? `:quit` doesn't throw an exception, it simply terminates the running Nvim process as well as the terminal process. What are we testing there?

Reply:
It at least is a regression test (though it should go at the end of the test, of course, since the test nvim will exit), which adds coverage/when we fix :quit to warn when the terminal-job is running.